code/game/objects/effects/effect_system/smoke.dm

```diff
@@ -94,6 +94,9 @@
 			M.smokecloak_off()
 
 /obj/effect/particle_effect/smoke/proc/apply_smoke_effect(turf/T)
+	if(!T && !QDELING(src))
```

Comment:
Same here for the `if(!T)` check. This could only happen if there's somehow a `/obj/effect/particle_effect/smoke` in null space.
`apply_smoke_effect()` is being called on three different procs, you might want to check that logic.

Reply:
What I'm doing here is specifically stop the proc from runtiming when the source ever happens to be in null space. As I said again, I'm placing one safety check inside the proc rather than three outside.
I could suppose the reason why the src was in null space due to the smoke spread affixed location, but very finnicky hazards due to the smoke being slow to set up that'll be patched on the way.
Plus the runtime report gives almost no useful information, originating from the processing subsystem ignite().

Reply:
Having three checks outside the proc instead of one inside is not necessarily a bad thing in DM, proc overhead considered. On other languages the alternative approach might make more sense.